sdk/storage/AzureStorageBlob/Source/Data/BlobTransfer+CoreDataClass.swift

```diff
@@ -55,7 +55,7 @@ public class BlobTransfer: NSManagedObject, Transfer {
 
     public var state: TransferState {
         get {
-            let currState = TransferState(rawValue: rawState) ?? .unknown
```

Comment:
This will cause the app to crash if a bad string value is provided. I think this change should be reverted. I would rather have the app not crash and allow the user to detect an abnormal condition that to use `!` and crash the app.

Reply:
I went back and forth on this, we can discuss in person - my reasoning was that the raw value comes from the enum, and is persisted to the database as a string, so we should only ever get back a valid value from the database. The only way I can see this failing is if the database is edited manually. 

This is why I removed `unknown` - the only way this could occur is if something massively screwed up in our CoreData database, and I didn't want to expose this in our public API. I feel like a customer seeing that a transfer can either be an upload, download, or "unknown" would immediately ask "Why could this be unknown? Your library owns this type and owns the job of persisting it to/from disk."

After some thought I can see that keeping "unknown" -could- be valuable in one specific edge case - that is, if we add more values to this enum in the future, and then an app uses the new version of the library (so a new value gets persisted to disk) but then later somehow an older version of the app using an older version of our library tries to load the record from disk and gets a value it's not expecting. I can't imagine this would be common though, and in that case I'd argue that -some- sort of failure is the right thing to do, rather than just return "unknown".

Reply:
Per our discussion, let's make these `Type!` and validate the nil aspect in loadContext.